autogen/variables.tf.tmpl

```diff
@@ -184,6 +184,18 @@ variable "autoscaling_lb" {
   default     = []
 }
 
+variable "autoscaling_scale_in_control" {
+  description = "Autoscaling, scale-in control block as single element array. https://www.terraform.io/docs/providers/google/r/compute_autoscaler.html#scale_in_control"
```

Comment:
For modules, we generally try to avoid using single-element arrays (as they are confusing).

I suggest we accept an input of this form:
```
  scale_in_control = {
    time_window_sec = number
    fixed_replicas = number
    percent_replicas = number
  }
```

Then make the whole block conditional on whether fixed_replicas or percent_replicas are set.

Reply:
I was trying to replicate the resource structure and noticed single-element array was the workaround for other fields like `cpu_utilization`. But I kind of dislike these artificial arrays, so happy to change the approach =)

Reply:
Yeah, sometimes we have gotten lazy with the implementation (which is easier with the single-element arrays) but generally prefer to avoid them where possible.